lib/MusicBrainz/Server/Data/WebService.pm

```diff
@@ -205,7 +205,7 @@ sub xml_search
     if ($query =~ /^\s*$/)
     {
         return {
-            error => "Must specify a least one parameter (other than 'limit', 'offset' or empty 'query') for collections query.",
+            error => "You submitted a blank query. You must include a non-blank 'query' parameter with your search.",
```

Comment:
It would be nice to additionally link to https://musicbrainz.org/doc/Development/XML_Web_Service/Version_2/Search

Reply:
I guess since this is a WS error, "link to" literally means including something like "See https://musicbrainz.org/doc/Development/XML_Web_Service/Version_2/Search for more info." at the end of the error line?

Reply:
It would probably be more convenient if the URL ended the line as 
in [XML WS error message](https://github.com/metabrainz/musicbrainz-server/blob/v-2019-06-03/lib/MusicBrainz/Server/WebService/XMLSerializer.pm#L1329): `For more info, please see: https://musicbrainz.org/doc/Development/XML_Web_Service/Version_2/Search`

I just noticed that JSON WS documentation doesn’t even mention search.

Reply:
And now I realize that the assumption “You submitted a blank query.” might be wrong because `fmt=json` may have been submitted as query string.

Reply:
Not sure I understand, can you given an example of `fmt=json` being submitted as a query string?

Reply:
Here is an example: https://musicbrainz.org/ws/2/artist?fmt=json

Reply:
Terminology conflict! I thought you meant the literal `query` parameter.

But that's what the message is supposed to mean by "blank query," that no search `query` was given, not that no query parameters were given.

Reply:
@mwiencek: So the message is still unclear. Another issue is it wrongly assumes the request is a search request missing its `query` query parameter (because other valid cases have been checked beforehand) whereas it could also be a browse request missing its MBID subpath or a lookup request missing its entity query parameter. Would a more general error message do?
`Your request is missing either an MBID or a query parameter.`

@reosarevok: I pushed a commit to add doc link to every JSON WS error, not just this one, on the model of XML WS error (which is updated to https in another commit). Can you please review it?

Reply:
@yvanzo Your commits seem reasonable and work locally :) 

It assumes it's a search request because this is xml_search. If it's not supposed to be a search, then it shouldn't hit xml_search, should it? :)

Reply:
That's https://github.com/metabrainz/musicbrainz-server/pull/1027 - once that PR is updated, browse requests won't end up here (that and anything else, including broken lookups, should return `The given parameters do not match any available query type for the artist resource.`). So this'll be specific to searches.

Not sure how to make the message more clear. `You submitted a blank search query. You must include a non-blank 'query=' parameter with your search.`?

Reply:
Alright, let’s leave that other issue to the other PR. Your latest suggestion is fine by me.

Reply:
Updated wording to @mwiencek's suggestion, which is nice :)